lib/bolt/cli.rb

```diff
@@ -83,186 +84,232 @@ class CLI
 Usage: bolt file <action> [options]
 
 Available actions are:
     upload <src> <dest>              Upload local file <src> to <dest> on each node
 
 Available options are:
-HELP
+      HELP
 
-    COMMANDS = { 'command' => %w[run],
-                 'script'  => %w[run],
-                 'task'    => %w[show run],
-                 'plan'    => %w[show run],
-                 'file'    => %w[upload] }.freeze
-    TRANSPORTS = %w[ssh winrm pcp].freeze
-
-    attr_reader :parser, :config
-    attr_accessor :options
-
-    def initialize(argv)
-      Bolt::Logger.initialize_logging
-      @argv    = argv
-      @options = {
-        nodes: []
-      }
-      @config = Bolt::Config.new
-
-      # parse mode and object, use COMMANDS as a whitelist
-      @options[:mode] = argv[0] if COMMANDS.keys.any? { |mode| argv[0] == mode }
-      @options[:object] = argv[1] if COMMANDS.values.flatten.uniq.any? { |object| argv[1] == object }
-      @parser = create_option_parser(@options)
-      @logger = Logging.logger[self]
-    end
+      # A helper mixin for OptionParser::Switch instances which will allow
+      # us to show/hide particular switch in the help message produced by
+      # the OptionParser#help method on demand.
+      module SwitchHider
+        attr_accessor :hide
 
-    def create_option_parser(results)
-      parser = OptionParser.new('') do |opts|
-        unless results[:mode] == 'plan'
-          opts.on('-n', '--nodes NODES',
-                  'Node(s) to connect to in URI format [protocol://]host[:port] (Optional)',
-                  'Eg. --nodes bolt.puppet.com',
-                  'Eg. --nodes localhost,ssh://nix.com:2222,winrm://windows.puppet.com',
-                  "\n",
-                  '* NODES can either be comma-separated, \'@<file>\' to read',
-                  '* nodes from a file, or \'-\' to read from stdin',
-                  '* Windows nodes must specify protocol with winrm://',
-                  '* protocol is `ssh` by default, may be `ssh` or `winrm`',
-                  '* port defaults to `22` for SSH',
-                  '* port defaults to `5985` or `5986` for WinRM, based on the --[no-]ssl setting') do |nodes|
-            results[:nodes] << get_arg_input(nodes)
-          end
+        def summarize(*args)
+          return self if hide
+          super
         end
-        opts.on('-u', '--user USER',
-                "User to authenticate as (Optional)") do |user|
-          results[:user] = user
+      end
+
+      def initialize(options)
+        super()
+
+        @options = options
+
+        @nodes = define('-n', '--nodes NODES',
+                        'Node(s) to connect to in URI format [protocol://]host[:port] (Optional)',
+                        'Eg. --nodes bolt.puppet.com',
+                        'Eg. --nodes localhost,ssh://nix.com:2222,winrm://windows.puppet.com',
+                        "\n", # the help message won't print if replaced with an empty string
```

Comment:
I'm not sure what this is trying to indicate.

Reply:
I'll make the comment clearer.